### PR TITLE
Remove unnecessary ESProducer from LegacyPFClusterProducer

### PR DESCRIPTION
**Pull request by jsamudio**
#### PR description:

This PR removes `pfClusterParamsESProducer` from `LegacyPFClusterProducer` as it is not used in the producer. Configurations for offline workflows and `customizeHLTforAlpaka.py` are updated to reflect this change.

#### PR validation:

Tested on workflow `12434.423` which uses the HLT customization and HCAL only configuration for this producer.

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Not a backport.

@hatakeyamak @fwyzard @waredjeb

### CODE REVIEW

**Comment by mmusich** [comment]
enable gpu

**Comment by Martin-Grunewald** [comment]
please test

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by mmusich** [comment]
@cmsbuild, please abort

**Comment by mmusich** [comment]
@cmsbuild, please test

File: RecoParticleFlow/PFClusterProducer/plugins/LegacyPFClusterProducer.cc [hunk 1]
```diff
@@ -68,7 +66,6 @@ class LegacyPFClusterProducer : public edm::stream::EDProducer<> {
     edm::ParameterSetDescription desc;
     desc.add<edm::InputTag>("src");
     desc.add<edm::InputTag>("PFRecHitsLabelIn");
-    desc.add<edm::ESInputTag>("pfClusterParams");
```
Comment (mmusich) [inline RecoParticleFlow/PFClusterProducer/plugins/LegacyPFClusterProducer.cc:71]: relaying an observation from @missirol, if this module has to be parsed in confDB we need to provide defaults in the `fillDescription`

**Comment by mmusich** [comment]
type pf

File: RecoParticleFlow/PFClusterProducer/plugins/LegacyPFClusterProducer.cc (hunk 1 above)
Comment (Martin-Grunewald) [reply RecoParticleFlow/PFClusterProducer/plugins/LegacyPFClusterProducer.cc:71]: Indeed, after the removal of ```pfClusterParams``` we still need default values in fillDescriptions added for:
```
InputTag src
InputTag PFRecHitsLabelIn
InputTag recHitsSource
```
Empty InputTags are fine, but the values must be there.

**Comment by Martin-Grunewald** [comment]
@jsamudio 

Could you please urgently address the above change? We'd need this (and the 14_0 bp) in the next 14_0 release. Thanks!

**Comment by Martin-Grunewald** [comment]
please test

**Comment by mmusich** [comment]
please test with https://github.com/cms-sw/cmssw/pull/44221

**Comment by Martin-Grunewald** [comment]
+1

**Comment by Martin-Grunewald** [comment]
@cms-sw/reconstruction-l2 can you please sign this PR as well, as you have signed already the backport to 14_0? Thanks!
@jfernan2 @mandrenguyen

**Comment by Martin-Grunewald** [comment]
urgent
(It's backport is neede for the next 14_0 release).

**Comment by mmusich** [comment]
@cms-sw/reconstruction-l2 please *do* sign now. This is a showstopper for HLT integration

**Comment by antoniovilela** [comment]
@cms-sw/reconstruction-l2
Please do review & sign as soon as possible.
Thanks.

**Comment by mandrenguyen** [comment]
+1

**Comment by antoniovilela** [comment]
+1

**Pull request merged**